workspace/apps/pidp/src/app/features/access/pages/bc-provider-edit/bc-provider-edit.page.ts

```diff
@@ -108,6 +129,39 @@ export class BcProviderEditPage
     this.navigationService.navigateToRoot();
   }
 
+  public onResetMfa(): void {
+    const data: DialogOptions = {
+      title: 'Multi-factor authentication (MFA) reset',
+      bottomBorder: false,
+      titlePosition: 'center',
+      bodyTextPosition: 'center',
+      component: HtmlComponent,
+      data: {
+        content:
+          'At your request we are about to reset your MFA, this action cannot be undone. Are you sure you want to do this?',
+      },
+      imageSrc: '/assets/images/online-marketing-hIgeoQjS_iE-unsplash.jpg',
+      imageType: 'banner',
+      width: '31rem',
+      height: '24rem',
+      actionText: 'Continue',
+      actionTypePosition: 'center',
+    };
+    this.dialog
+      .open(ConfirmDialogComponent, { data })
+      .afterClosed()
+      .pipe(
+        exhaustMap((result) =>
+          result ? this.resetMfa(this.partyService.partyId) : EMPTY,
+        ),
+      )
+      .subscribe();
+  }
+
+  public onLearnMore(): void {
+    this.router.navigateByUrl(FaqRoutes.BASE_PATH);
```

Comment:
can we place the navigation logic in one helper function in switch case and redirect based on the parameter passed.
So that we don't redefine the same navigation function across components. If any changes comes changing in the one helper function will be scalable.

Reply:
I think what you are proposing would work for components with parent - child relationship, unless I misunderstand what you mean.

Reply:
I mean 
navigateHelper(pageName)  {
const path = constant_paths.pageName;
 router.navigateByUrl(path);
}

In constants we define paths like :
constant_paths = {
homepage: FaqRoutes.BASE_PATH,
profilepage: FaqRoutes.PROFILE_PATH 
}

Reply:
@kakarlavinodkumar please create ticket to address :)